static/src/javascripts/projects/common/modules/navigation/newHeaderNavigation.js

```diff
@@ -1,129 +1,126 @@
 // @flow
 
-import qwery from 'qwery';
 import fastdom from 'fastdom';
 import ophan from 'ophan/ng';
 import userAccount from 'common/modules/navigation/user-account';
 
-const html = qwery('html')[0];
-const menuItems = qwery('.js-close-nav-list');
+const sidebar = document.getElementById('main-menu');
 const enhanced = {};
 
-const weShouldEnhance = (checkbox: HTMLInputElement): ?boolean =>
-    checkbox && !enhanced[checkbox.id] && !checkbox.checked;
+const closeSidebarSections = (targetItem?: HTMLElement): void => {
+    const sections = [...document.querySelectorAll('.js-close-nav-list')];
 
-const closeAllOtherPrimaryLists = (targetItem?: HTMLElement): void => {
-    menuItems.forEach(item => {
-        if (item !== targetItem) {
-            item.removeAttribute('open');
+    sections.forEach(section => {
+        if (section !== targetItem) {
+            section.removeAttribute('open');
         }
     });
 };
 
-const removeOrderingFromLists = (): void => {
-    const mainListItems = qwery('.js-navigation-item');
+const toggleSidebar = (trigger: HTMLElement): void => {
+    const documentElement = document.documentElement;
+    const sidebarToggle = document.querySelector('.js-change-link');
+    const openClass = 'new-header__nav__menu-button--open';
+    const globalOpenClass = 'nav-is-open';
+    const isOpen = trigger.getAttribute('aria-expanded') === 'true';
 
-    mainListItems.forEach(item => {
-        const listItem = item;
+    if (!sidebar || !sidebarToggle) {
```

Comment:
We just ran an experiment and removed the null check. Interestingly, Flow complains even without passing the sidebar in as a parameter; it is able to infer the fact that sidebar is nullable in its current scope.

That said I do agree that we should try to keep variables scoped as tightly as possible and functions self-contained.